services/rsources/handler.go

```diff
@@ -803,3 +809,18 @@ func (sh *sourcesHandler) Monitor(ctx context.Context, lagGauge, replicationSlot
 		}
 	}
 }
+
+func withAdvisoryLock(ctx context.Context, db *sql.DB, lockId int64, f func(tx *sql.Tx) error) error {
+	tx, err := db.BeginTx(ctx, nil)
+	if err != nil {
+		return fmt.Errorf("failed to start transaction: %w", err)
+	}
+	defer func() { _ = tx.Rollback() }()
+	if _, err := tx.ExecContext(ctx, `SELECT pg_advisory_xact_lock($1)`, lockId); err != nil {
+		return fmt.Errorf("error while acquiring advisory lock: %w", err)
```

Comment:
[minor] For readability, given that the can be wrapped multiple times, error strings in go should not contain words `error` or `failed`. Consider the error message you want to have after logging an error wrapped multiple times:

`ERROR: error while starting service: error running the rsources migration: error while acquiring advisory lock: ...` vs

`ERROR: starting service: running the rsources migration: acquiring advisory lock: ...`

```suggestion
		return fmt.Errorf("acquiring advisory lock: %w", err)
```